chaosaws/ec2/actions.py

```diff
@@ -99,14 +99,105 @@ def stop_instances(instance_ids: List[str] = None, az: str = None,
                                    force=force, client=client)
 
 
+def terminate_instance(instance_id: str = None, az: str = None,
+                       filters: List[Dict[str, Any]] = None,
+                       configuration: Configuration = None,
+                       secrets: Secrets = None) -> List[AWSResponse]:
+    """
+    Terminates a single EC2 instance.
+
+    An instance may be targeted by specifying it by instance-id. If only the
+    availability-zone is provided, a random instances in that AZ will be
+    selected and terminated. For more control, please reference the available
+    filters found:
+    https://bit.ly/2Sv9lmU
+    """
+
+    if not any([instance_id, az, filters]):
+        raise FailedActivity('To terminate an EC2, you must specify the '
+                             'instance-id, an Availability Zone, or provide a '
+                             'set of filters')
+
+    if az and not any([instance_id, filters]):
+        logger.warning(
+            'Configuration will terminate a random instance in %s.' % az)
+
+    client = aws_client('ec2', configuration, secrets)
+    if not instance_id:
+        filters = deepcopy(filters) or []
+
+        if az:
+            filters.append({'Name': 'availability-zone', 'Values': [az]})
+            logger.debug('Looking for instances in AZ: %s' % az)
+
+        # Randomly select an instance
+        instance_types = pick_random_instance(filters, client)
+
+        if not instance_types:
+            raise FailedActivity(
+                'No instances found matching filters: %s' % str(filters))
+
+        logger.debug('Instance selected: %s' % str(instance_types))
+    else:
+        instance_types = get_instance_type_by_id([instance_id], client)
+
+    return terminate_instances_any_type(instance_types, client)
+
+
+def terminate_instances(instance_ids: List[str] = None, az: str = None,
+                        filters: List[Dict[str, Any]] = None,
+                        configuration: Configuration = None,
+                        secrets: Secrets = None) -> List[AWSResponse]:
+    """
+    Terminates multiple EC2 instances
+
+    A set of instances may be targeted by providing them as the instance-ids.
+
+    WARNING: If  only an Availability Zone is specified, all instances in
```

Comment:
It's interesting because we have an issue around those particular actions that can have a larger effect than anticipated.

https://github.com/chaostoolkit/chaostoolkit/issues/94

Reply:
I've added an `input()` call for the operations involving blind AZ operations. Not sure if that's the solution you had in mine.